app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,752 @@
+                body: {
+                    message: string `Input email is not a valid WSO2 email address: ${email}`
+                }
+            };
+        }
+        types:Employee|error? employee;
+        if userInfoCache.hasKey(email) {
+            return userInfoCache.get(email).ensureType();
+        }
+
+        employee = database:getEmployee(email);
+        if employee is error {
+            string customError = string `Error getting employee basic information for ${email}`;
+            log:printError(customError, employee);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+        if employee is () {
+            log:printDebug(string `${types:NOT_FOUND_CUSTOM_ERROR}: ${email}`);
+            return <http:NotFound>{
+                body: {
+                    message: string `${types:NOT_FOUND_CUSTOM_ERROR}: ${email}`
+                }
+            };
+        }
+        return employee;
+    }
+
+    # Get basic information of employees.
+    #
+    # + filters - Filter objects containing the filter criteria for the query
+    # + 'limit - The maximum number of employees to return
+    # + offset - The number of employees to skip before starting to collect the result set
+    # + return - Basic information of the employees or an error
+    resource function post employees(types:EmployeeFilter? filters, int? 'limit, int? offset)
+        returns types:Employee[]|http:BadRequest|http:InternalServerError|http:NotFound {
+
+        types:Employee[]|error employees =
+            database:getEmployees(filters ?: {}, 'limit ?: types:DEFAULT_LIMIT, offset ?: 0);
+        if employees is error {
+            string customError = "Error getting employee basic information for employees";
+            log:printError(customError, employees);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+        return employees;
+    }
+
+    # Organization structure data retrieval.
+    #
+    # + return - Internal Server Error or Organization structure data
+    resource function get org\-structure() returns types:OrgStructure|error {
+
+        if orgStructureCache.hasKey("orgStructure") {
+            return <types:OrgStructure>check orgStructureCache.get("orgStructure");
+        }
+
+        types:OrgStructure orgStructure = {
+            businessUnits: check database:getOrgDetails(filter = {}, 'limit = types:DEFAULT_LIMIT, offset = 0),
+            companies: check database:getCompanies(filter = {}, 'limit = types:DEFAULT_LIMIT, offset = 0),
+            careerPaths: check database:getCareerFunctions(filter = {}, 'limit = types:DEFAULT_LIMIT, offset = 0),
+            employmentTypes: [types:PERMANENT, types:INTERNSHIP, types:CONSULTANCY]
+        };
+        check orgStructureCache.put("orgStructure", orgStructure);
```

Comment:
Better to use a constant (or an enum) for these keys instead of hard coding it here.
```suggestion
        check orgStructureCache.put("orgStructure", orgStructure);
```

Reply:
ACK